app/services/foreman/plugin/role_lock.rb

```diff
@@ -7,33 +7,39 @@ def initialize(plugin_id)
         @plugin_id = plugin_id
       end
 
-      def register_role(name, permissions, role_registry)
+      def register_role(name, permissions, role_registry, description = nil)
         User.as_anonymous_admin do
           Role.transaction do
-            role = process_role name, permissions
+            role = process_role name, permissions, description
             role_registry.role_ids << role.id
           end
         end
       end
 
-      def process_role(name, permissions)
+      def process_role(name, permissions, description = nil)
         role = Role.find_by :name => name
-        if role && role.origin && role.permission_diff(permissions).present?
-          return update_plugin_role_permissions role, permissions
-        end
-        if role && role.permission_diff(permissions).empty?
-          role.update_attribute :origin, @plugin_id if role.origin.empty?
-          return role
+        if role
+          role.description = description
+
+          if role.origin && role.permission_diff(permissions).present?
+            return update_plugin_role_permissions role, permissions
+          end
+
+          if role.origin.empty? && role.permission_diff(permissions).empty?
+            role.update_attribute :origin, @plugin_id
+            return role
```

Comment:
This is becoming a bit complicated, maybe break up into functions?

Reply:
there's not much we can extract, since two conditions are using return, we'd need to double return which wouldn't help the readability that much, I'd prefer to keep it as it is